### PR TITLE
Add monotonicTime to MonadTime

### PR DESCRIPTION
**Pull request by jsynacek**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# monad-time-0.3.2.0 (2022-07-04)
```
Comment (jsynacek) [inline CHANGELOG.md]: Hmm, should this be 0.4 instead?

File: monad-time.cabal [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name:                monad-time
-version:             0.3.1.0
+version:             0.3.2.0
```
Comment (jsynacek) [inline monad-time.cabal]: Hmm, should this be 0.4 instead?

File: CHANGELOG.md (hunk 1 above)
Comment (arybczak) [reply CHANGELOG.md]: Yes.

**Comment by arybczak** [comment]
I don't understand. `MonadTime` would give you a way to get both "current" time and monotonic time, I don't see a problem/conflict of interests here :thinking:

**Comment by jsynacek** [comment]
I went by the comments that say "Class of monads which carry the notion of the current time.". But, if you think `monotonicTime` should be added to `MonadTime`, let's do it then. I just need to know what to do with the `MonadTime (ReaderT UTCTime m)` instance, because it makes no sense to add `monotonicTime` to it.

**Comment by arybczak** [comment]
> Class of monads which carry the notion of the current time.

Let's update it to "class of monads which make it possible to measure time". I think that covers both cases.

> I just need to know what to do with the MonadTime (ReaderT UTCTime m) instance, because it makes no sense to add monotonicTime to it.

Hmm, good point.

I didn't even know such instance was added, apparently in #2. I think it's fine to remove it if we bump the version. I'm not a fan of this instance :thinking:

File: README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
 # monad-time [![Hackage version](https://img.shields.io/hackage/v/monad-time.svg?label=Hackage)](https://hackage.haskell.org/package/monad-time) [![Build Status](https://secure.travis-ci.org/scrive/monad-time.svg?branch=master)](http://travis-ci.org/scrive/monad-time)
 
-A `MonadTime` type class for monads which carry the notion of the
-current time.
+`MonadTime` and `MonadMonotonicTime` type classes for monads which carry the notion of the
```
Comment (arybczak) [inline README.md]: Needs update.

File: monad-time.cabal [hunk 4]
```diff
@@ -17,7 +17,8 @@ build-type:          Simple
 cabal-version:       >=1.10
 extra-source-files:  README.md CHANGELOG.md
 tested-with:         GHC == 7.4.2, GHC == 7.6.3, GHC == 7.8.4, GHC == 7.10.3,
-                     GHC == 8.0.2, GHC == 8.2.2, GHC == 8.4.4, GHC == 8.6.2
+                     GHC == 8.0.2, GHC == 8.2.2, GHC == 8.4.4, GHC == 8.6.2,
+                     GHC == 8.10.7, GHC == 9.2.3
```
Comment (arybczak) [inline monad-time.cabal:21]: 9.0.2 is missing. Also, you can drop versions older than 8.8.

**Review by arybczak — APPROVED** [review]

**Pull request merged**